[ADD] Employees: Onboarding (#13129)
odoo/documentation: content/applications/hr/employees/onboarding.rst

to be successful in their role, and ensures a smooth transition to their new job.

View onboarding plan
====================

Before onboarding can begin, it is recommended to check the default onboarding plan that comes
preconfigured with the **Employees** app. To view the current default plan, navigate to
:menuselection:`Employees app --> Configuration --> Activity Plan`.

The two default plans are :guilabel:`Offboarding` and :guilabel:`Onboarding`. Click
:guilabel:`Onboarding` to view the detailed onboarding plan form.

.. _onboarding/plan-form:

The plan form displays the following information:

- :guilabel:`Plan Name`: the specific name for the plan.
- :guilabel:`Model`: specifies where this plan can be used. In this case, in the **Employees** app.
- :guilabel:`Department`: if left blank (the default setting) this plan is available for all
  departments. The plan can be department-specific by selecting the department using the drop-down
  menu.
- :guilabel:`Activities To Create`: this tab lists all the steps that must be carried out in order
  to complete the onboarding plan. The steps appear in a list view, and each step displays the
  following information:

>   - :guilabel:`Activity Type`: the specific activity for the step. The default options are :guilabel:`Email`,

Author: What's odd is that I did run make review- but it never catches this line, even though it looks like it's 110 long. I just checked again - an "no issues" is reported.

plan needed is only for a specific department, then a new plan should be created, specifically for
that department.

To modify the default plan, first navigate to :menuselection:`Employees app --> Configuration -->
Activity Plan`, then click on :guilabel:`Onboarding`.

To modify a step, click on the step and an :guilabel:`Open: Activities` pop-up window appears. Make
any desired modifications to the step, then click :guilabel:`Save` to accept the changes and close
the pop-up window.

To add a new step, click :guilabel:`Add a line` at the bottom of the listed activities in the
:guilabel:`Activities To Create` tab, and a blank :guilabel:`Create Activities` pop-up window
appears. Enter all the information in the pop-up window, then click :guilabel:`Save & Close` if
there are no their steps to add, or click :guilabel:`Save & New` if more steps are needed.

Configure all the desired steps for the onboarding plan.

Create onboarding plan
======================

For some companies, specific onboarding plans may be necessary for some departments. For these
cases, a new department-specific onboarding plan may be needed.

> To create a new onboarding plan, first navigate to :menuselection:`Employees app -->

Author: Ahh - I know why - I tried to move the word, but it didn't render right for some reason (the arrows were messed up) but when I moved it back it worked. But today I moved it back again, and it rendered fine!

Before onboarding can begin, it is recommended to check the default onboarding plan that comes
preconfigured with the **Employees** app. To view the current default plan, navigate to
:menuselection:`Employees app --> Configuration --> Activity Plan`. Click :guilabel:`Onboarding` to
view the detailed onboarding plan form.

.. _onboarding/plan-form:

The plan form displays the following information:

- :guilabel:`Plan Name`: the specific name for the plan.
- :guilabel:`Model`: specifies where this plan can be used. In this case, in the **Employees** app.
- :guilabel:`Department`: if left blank (the default setting) this plan is available for all
  departments. The plan can be department-specific by selecting the department using the drop-down
  menu.
- :guilabel:`Activities To Create`: this tab lists all the steps that must be carried out in order
  to complete the onboarding plan. The steps appear in a list view, and each step displays the
  following information:

  - :guilabel:`Activity Type`: the specific activity for the step. The default options are :guilabel:`Email`,
    :guilabel:`Call`, :guilabel:`Meeting`, :guilabel:`To-Do`, or :guilabel:`Upload Document`. If the

>     **Sign** app is installed, a :guilabel:`Request Signature` option is available.

Author: Looks like it works fine/no issues now (for the comment below - I can't comment on it for some reason! But it works now, no issues, and no character over the limit (I was having issues with it previously for whatever reason).

  - :guilabel:`Summary`: a short description of the step.

>   - :guilabel:`Assignment`: the user assigned to perform the activity. The default options are:
>     :guilabel:`Ask at launch`, :guilabel:`Default user`, :guilabel:`Coach`, :guilabel:`Manager`, and
>     :guilabel:`Employee`. If the **Fleet** app is installed, a :guilabel:`Fleet Manager` option is
>     available.
> 
>     .. note::
>        The selection for the :guilabel:`Assignment` role is in relation to the employee. If
>        :guilabel:`Coach` is selected, the employee's coach is assigned to the activity.
> 
>        If :guilabel:`Default user` is selected, an :guilabel:`Assigned to` field appears. Using the
>        drop-down menu, select the user who will always be assigned this activity.

Reviewer: For this section, I think it'd be better to define what the different assignment options do in the database, and who they are. For example:

```suggestion
  - :guilabel:`Assignment`: chooses who completes the activity, relative to the new hire.

    - :guilabel:`Ask at launch`: [NEEDS INFO: what happens in Odoo when we pick this? when we launch
      the onboarding plan, is there a pop-up that asks you to pick a user?]

    -  :guilabel:`Default user`: choose a user who always handles this activitiy. Defined in the
       :guilabel:`Assigned to` field.

    - :guilabel:`Manager`: [needs info]
    - :guilabel:`Coach`:  assigns the employee's coach as defined on the employee record.
    - :guilabel:`Employee`: the new hire completes the activity.
    - :guilabel:`Fleet Manager`: [needs info]
```

Author: I like this version a lot better- great suggestion!

  - :guilabel:`Assigned to`: this field remains blank unless :guilabel:`Default user` is selected
    for the :guilabel:`Assignment` field. If :guilabel:`Default user` is selected, this field is
    populated with the selected user.

>   - :guilabel:`Document to sign`: the corresponding document requiring a signature.

Reviewer: will have to delete/move this if the above suggestion is accepted!

>   - :guilabel:`Interval`: the time when the activity is due.

Reviewer: how about adding an example, like a week of onboarding?

>   - :guilabel:`Interval`: the time when the activity is due.

Reviewer: due makes it sound like a deadline instead of a process (:
```suggestion
  - :guilabel:`Interval`: the time when the activity is active.
```

       is set to :guilabel:`days`. The :guilabel:`Trigger` is :guilabel:`Before Plan Date`.

       .. image:: onboarding/activity-plan.png
          :alt: An activity configured to assign a laoptop the day before an employee starts work.

Onboarding plan steps
---------------------

The default :guilabel:`Onboarding` plan includes three default steps. All steps are
:guilabel:`To-Do` activities, and are scheduled for the day the onboarding plan is launched
(:guilabel:`0 days Before Plan Date`).

- :guilabel:`Setup IT Materials`: the manager must gather and configure all IT materials.
- :guilabel:`Plan Training`: the manager must plan the training for the new employee.
- :guilabel:`Training`: the new employee must complete the training planned by the manager.

.. image:: onboarding/onboarding.png
   :alt: The three default steps in the Onboarding plan.

Modify onboarding plan
======================

Modifications can be made to the default onboarding plan to accommodate any company's onboarding
needs. Every company has different requirements, thus it is necessary to add the required steps to

> the onboarding plan.

Reviewer: Suggested rewrite for brevity!

A single onboarding plan works only if the flow works for the entire company. To modify existing onboarding plans to fit the needs of specific departments or teams, go to... (navigational instructions)

.. note::
   If the onboarding plan is universal, add or modify the default onboarding plan. If
   department-specific onboarding plans are needed, :ref:`create a new onboarding plan
   <employees/create-plan>`.

To modify the default plan, first navigate to :menuselection:`Employees app --> Configuration -->
Activity Plan`, then click on :guilabel:`Onboarding`.

To modify a step, click on the step and an :guilabel:`Open: Activities` pop-up window appears. Make
any desired modifications to the step, then click :guilabel:`Save` to accept the changes and close
the pop-up window.

To add a new step, click :guilabel:`Add a line` at the bottom of the listed activities in the
:guilabel:`Activities To Create` tab, and a blank :guilabel:`Create Activities` pop-up window
appears. Enter all the information in the pop-up window, then click :guilabel:`Save & Close` if

> there are no their steps to add, or click :guilabel:`Save & New` if more steps are needed.

Reviewer: ```suggestion
there are no other steps to add, or click :guilabel:`Save & New` if more steps are needed.
```

.. _employees/create-plan:

> Create onboarding plan

Reviewer: I'm not sure if we need to go into detail for this section, since it's quite similar to the modify one. Are there any steps that are very similar that we only need to write once?

In this create onboarding plan, section, we can just summarize the same steps and link to sections that go over it in the "modify" section, and go into detail only when we have a different step. That way, we can keep this doc short and sweet (:

======================

Some companies require different onboarding plans, when there are department-specific onboarding

> procedures that do not apply ot the whole company. For these cases, a new department-specific

Reviewer: ```suggestion
procedures that do not apply to the whole company. For these cases, a new department-specific
```

Activities` pop-up window appears. Enter all the information in the pop-up window, then click
:guilabel:`Save & Close` if there are no other steps to add, or click :guilabel:`Save & New` if more
steps are needed.

Configure all the desired steps for the onboarding plan.

.. example::
   A company specializing in the manufacturing and selling of outdoor metal furniture may have a
   large factory that produces the products, and a separate sales office. This company may have two
   separate onboarding plans, one for factory workers, and one for office workers.

   The onboarding plan for the factory workers is set for the :guilabel:`Manufacturing` department,
   and includes specialized tasks relating to factory jobs. These include gathering the new
   employees uniform and safety gear, assigning a safety course, emailing their team about the new
   hire, going over benefits, and more.

   .. image:: onboarding/factory-onboarding.png
      :alt: An onboarding plan configured for factory workers.

Launch onboarding plan
======================

After an employee has been hired, and their employee profile :ref:`is created

> <recruitment/new-employee>`, click the :guilabel:`Launch Plan` button on the employee profile, and a

Reviewer: ```suggestion
<recruitment/new-employee>`, navigate to the desired employee's profile (fill in the menselection instructions to go there), click the :guilabel:`Launch Plan` button on the employee profile, and a
```

blank :guilabel:`Launch Plan` pop-up window loads.

> Using the drop-down menu, select the desired onboarding plan in the :guilabel:`Plan` field. Then,

Reviewer: ```suggestion
In :guilabel:`Plan` field, choose the desired onboarding plan. Then,
```

using the calendar selector, set a date in the :guilabel:`Plan Date` field. This is typically the
employee's first day, but any date can be selected.

> The right-side of the :guilabel:`Launch Plan` pop-up window displays all the steps in the selected

Reviewer: ```suggestion
The right side of the :guilabel:`Launch Plan` pop-up window displays all the steps in the selected
```
